src/Collection/Collection.service.ts

```diff
@@ -22,18 +32,205 @@ import {
+      : this.getDCLCollection(dbCollection)
+  }
+
+  public async deleteCollection(collectionId: string): Promise<void> {
+    const collection = await this.getDBCollection(collectionId)
+
+    if (isTPCollection(collection)) {
+      await this.checkIfThirdPartyCollectionHasPublishedItems(
+        collection.id,
+        CollectionAction.DELETE
+      )
+    } else {
+      // If it's a DCL collection, we must check if it was already published
+      if (await this.isDCLPublished(collection.contract_address!)) {
+        throw new AlreadyPublishedCollectionError(
+          collectionId,
+          CollectionType.DCL,
+          CollectionAction.DELETE
+        )
+      }
+    }
+    if (this.isLockActive(collection.lock)) {
+      throw new LockedCollectionError(collection.id, CollectionAction.DELETE)
+    }
+    await Promise.all([
+      Collection.delete({ id: collection.id }),
+      // TODO: This should eventually be in the item's service
+      Item.delete({ collection_id: collection.id }),
+    ])
+  }
+
+  public async publishDCLCollection(
+    dbCollection: CollectionAttributes,
+    dbItems: ItemAttributes[]
+  ) {
+    const remoteCollection = await collectionAPI.fetchCollection(
+      dbCollection!.contract_address!
+    )
+
+    if (!remoteCollection) {
+      // This might be a problem with the graph lagging but we delegate the retry logic on the client
+      throw new UnpublishedCollectionError(dbCollection.id)
     }
 
-    const deadline = new Date(lock)
-    deadline.setDate(deadline.getDate() + 1)
+    const items: ItemAttributes[] = [...dbItems]
+    let remoteItems: ItemFragment[] = []
 
-    return deadline.getTime() > Date.now()
+    const isMissingBlockchainItemIds = dbItems.some(
+      (item) => item.blockchain_item_id == null
+    )
+
+    if (isMissingBlockchainItemIds) {
+      remoteItems = await collectionAPI.fetchItemsByContractAddress(
+        dbCollection!.contract_address!
+      )
+
+      const updates = []
+
+      for (const [index, item] of items.entries()) {
+        const remoteItem = remoteItems.find(
+          (remoteItem) => Number(remoteItem.blockchainId) === index
+        )
+        if (!remoteItem) {
+          throw new UnpublishedItemError(item.id)
+        }
+
+        items[index].blockchain_item_id = remoteItem.blockchainId
+        updates.push(
+          Item.update(
+            { blockchain_item_id: remoteItem.blockchainId },
+            { id: item.id }
+          )
+        )
+      }
+
+      await Promise.all(updates)
+    }
+
+    return {
+      collection: Bridge.mergeCollection(dbCollection!, remoteCollection),
+      items: await Bridge.consolidateItems(items, remoteItems),
+    }
+  }
+
+  /**
+   * It should receive the list of items to be operated on so it can decide where it should create or update the curations for them accordingly
+   * It will also receive a signed message and its signature, so it can check and store it in the SlotUsageCheque
```

Comment:
Would you mind rephrasing it so it explains what the method does?
```suggestion
   * Publishes a TP collection by storing the slots cheque and creating the curations for the items to be published.
```